src/python/pants/backend/go/util_rules/goroot.py

```diff
@@ -39,6 +39,11 @@ def is_compatible_version(self, version: str) -> bool:
         """Can this Go compiler handle the target version?"""
         return compatible_go_version(compiler_version=self.version, target_version=version)
 
+    def major_version(self, version: str) -> str:
+        _version_components = version.split(".")  # e.g. [1, 17] or [1, 17, 1]
+        major_version = f"{_version_components[0]}.{_version_components[1]}"
```

Comment:
I wonder if this part of the pants code might be the first place that introspects the go version, and thus the first place that encounter an _invalid_ version. 

For instance, if someone typoes as `117` or `1,17`, this code will crash with a non-obvious error (just a Python traceback). Maybe we can adjust slightly to be resilient to this:

```suggestion
        major_version = ".".join(_version_components[:2])
```

If the user has made an error, this code will just propagate that downstream for (hopefully) Go-specific tooling to give better errors.

(Other options would be emitting a better error message here, or validating the version early, e.g. in  `determine_go_mod_info`.)